src/platform_impl/windows/event_loop.rs

```diff
@@ -1903,11 +1903,24 @@ unsafe fn public_window_callback_inner<T: 'static>(
 
         old_physical_inner_rect
       };
-      let old_physical_inner_size = PhysicalSize::new(
+      let mut old_physical_inner_size = PhysicalSize::new(
         (old_physical_inner_rect.right - old_physical_inner_rect.left) as u32,
         (old_physical_inner_rect.bottom - old_physical_inner_rect.top) as u32,
       );
 
+      // When the "Show window contents while dragging" is turned off, there is no need to adjust the window size.
+      if !is_show_window_contents_while_dragging_enabled() {
+        subclass_input.send_event(Event::WindowEvent {
+          window_id: RootWindowId(WindowId(window.0)),
+          event: ScaleFactorChanged {
+            scale_factor: new_scale_factor,
+            new_inner_size: &mut old_physical_inner_size,
+          },
+        });
+        result = ProcResult::Value(LRESULT(0));
+        return;
```

Comment:
There are more events pending below. Do we have to return early here?

Reply:
I found that the main issue is that the `SetWindowPos` function was called at the end, and the new parameters passed in caused the window to enlarge. I moved the `return` to this position mainly so that the `ScaleFactorChanged` event could trigger normally. Of course, moving it to the end to make the judgment and directly calling the `SetWindowPos` function with the correct parameters sounds like a more prudent approach.

I'm not good at English, sorry.